nd2reader/parser.py

```diff
@@ -247,6 +247,28 @@ def _channel_offset(self):
         """
         return {channel: n for n, channel in enumerate(self.metadata["channels"])}
 
+    def _check_unwanted_bytes(self, image_group_data, image_data_start, height, width):
```

Comment:
Could you add some comments to this function that explain its purpose and the various steps you're taking?